SparkPost/2web2ui
# FAD-6306: Centralize most errors (4xx, etc)

_Refer to [FAD-6306](https://jira.int.messagesystems.com/browse/FAD-6306) for AC._

**Affected**

- Api Keys - creating, deleting, and updating
- Billing - updating contact info, updating plan, updating payment info
- Dashboard - verify sending domain
- Message events report including filtering
- In-App Support - creating
- Ip Pools - creating, deleting, updating
- Onboarding - create sending domain, create api key
- Profile (User) - updating info and password
- Recipient List - creating, deleting, searching, updating
- Sending Domains - creating, deleting, toggle default, updating, verifying, verifying with CName (the catch was removed, but a special case if `results.dns.cname_error` exists was left), verifying with DKIM, verifying with email
- SMTP - toggle engagement tracking
- Subaccounts - creating, updating
- Suppressions - deleting, get a list with filters, creating, uploading
- Templates - creating, deleting, previewing, publishing, updating
- Tracking domain - creating, deleting, updating, and verifying
- Users - deleting, inviting, updating
- Webhooks - creating, get refresh batches, deleting, showing (still redirects, but may remove), testing, updating

**Testing**

1. [Download Charles](https://www.charlesproxy.com/download/)
1. In "Proxy Settings", click "Use a dynamic port" in the "HTTP Proxy" panel.
1. In "Help" > "SSL Proxy", click "Install Charles Root Certificate".  [should open Keychain]
1. In Keychain, find the new certificate and double click it to open the info window
1. Expand "Trust" and change "When using this certificate" to "Always Trust"
1. Close the window to save the change
1. Start recording (the sequence tab is a better log than structure, especially with a filter)
1. Open production
1. Find a "CONNECT" API request (api.sparkpost.com), right click and click "SSL Proxying Enabled" from the menu
1. Reload the page [should see 200 GET requests with a path]
1. Open "Tools" > "Rewrite"
1. Import list of rewrites, [fad6306.xml.zip](https://github.com/SparkPost/2web2ui/files/1804295/fad6306.xml.zip)

**Todo**
- [x] Auto alert for SparkPost API errors
- [x] Auto alert for Zuora API errors
- [x] Remove manual alerting
